### ISSUE TITLE
Handle pronunciation tags properly in nested structures

### ISSUE THREAD

**Issue by Tc14Hd** [tatuylonen/wiktextract#115]
The pronunciation section of the word "incorporate" looks like this:

<img src="https://user-images.githubusercontent.com/56201517/153517984-1eb8677c-88a9-42e4-9f73-1f9d99e71c01.png" width="500">

Here is the corresponding source code:

```
====Pronunciation====
* {{q|verb}}
** {{a|Canada}} {{IPA|en|/ɪŋˈkɔɹpɚe(ɪ)t/}}
*** {{audio|en|en-ca-incorporate.ogg|Audio (CA)}}
** {{a|RP}} {{IPA|en|/ɪŋˈkɔː(ɹ).pəɹ.eɪt/}}
** {{a|US}} {{enPR|ĭnkôr'pərāt}}, {{IPA|en|/ɪŋˈkɔɹpɚeɪt/}}
* {{q|adjective}}
** {{a|Canada}} {{IPA|en|/ɪŋˈkɔɹpɚət/}}
** {{a|RP}} {{IPA|en|/ɪŋˈkɔː(ɹ).pəɹ.ət/}}
** {{a|US}} {{enPR|ĭnkôr'pərət}}, {{IPA|en|/ɪŋˈkɔɹpɚət/}}
```

As you can see, `{{q|verb}}` and `{{q|adjective}}` apply to multiple transcriptions, but right now they are ignored.

```
[
    {
        "ipa": "/\u026a\u014b\u02c8k\u0254\u0279p\u025ae(\u026a)t/",
        "tags": [
            "Canada"
        ]
    },
    {
        "ipa": "/\u026a\u014b\u02c8k\u0254\u02d0(\u0279).p\u0259\u0279.e\u026at/",
        "tags": [
            "Received-Pronunciation"
        ]
    },
    {
        "ipa": "/\u026a\u014b\u02c8k\u0254\u0279p\u025ae\u026at/",
        "tags": [
            "US"
        ]
    },
    ...
]
```

There should be an additional `"verb"` entry in the three `"tags"` arrays above. Can you fix this?

**Comment by tatuylonen** [comment]
This is a valid issue.  The fix is certainly doable, but requires some effort.  However, there are some other issues I want to prioritize first, so I'll leave this open for now and return to this later.  If you are willing to work on this, I would happy to merge a fix.  I think the best way to do it would be to make parse_pronunciation() in page.py recognize two-level lists and first parse tags from the first level and then parse each second-level list, and merge tags from the outer list to the pronunciation entries.

**Comment by Tc14Hd** [comment]
Thank you for considering this issue! I will probably not be able to fix this myself, so I will happily wait until you do.

**Comment by garfieldnate** [comment]
Not sure if I'm seeing exactly the same issue, but I was just inspecting the output for [produce](https://kaikki.org/dictionary/All%20languages%20combined/meaning/p/pr/produce.html) and the pronunciations for both the noun and verb form are present in both the noun and verb entries. I would expect the noun pronunciations not to be output at all for the verb senses and vice versa.

**Comment by kristian-clausal** [comment]
> On the plus side, the sense-specific pronunciations of [micrometer](https://kaikki.org/dictionary/All%20languages%20combined/meaning/m/mi/micrometer.html) are correctly separated, so I suppose this is only an issue with POS-specific pronunciations.

That's because the original wiktionary article separates the pronunciations into two etymologies. In the article for produce, it's all lumped together into one pronunciation under one etymology umbrella, and they're only distinguished by apparently ad hoc "Verb" and "Noun" lines, using [semicolon description list syntax](https://en.wikipedia.org/wiki/Help:Wikitext#Lists).

**Comment by tatuylonen** [comment]
Do we have any idea how common these are?  I'm beginning to think these should be fixed in Wiktionary if there aren't too many.

**Comment by kristian-clausal** [comment]
Fixed with 54a058bb77df73dfc3b638a750da87a0b91ed9c4, 05f0a5f2834655ba32b440cbd7445f242cd3c68a, and 23a7d8f20f44a7322eeb3482ba2d03a22cd9a4a7.

With these commits, which didn't break anything in our tests because I don't think we have tests for Pronunciation sections, the two issues are kind of kludged.

Now, all sublists in pronunciation sections are flattened into one list. Hierarchy doesn't *really* matter much here, from what I've seen, so most deeply nested lines are as valuable as any other, mainly because Pronunciation section templates are good at annotating themselves. The exception is when people do the thing with dividing up the list into several, which leads us to...

All lists that are one after another in pronounciation sections (like the one in [produce](https://en.wiktionary.org/wiki/produce)) are now joined together into one big list. This merges two kinds of divided lists: lists that are divided semantically and are (in wikitext code) one list with special lines that start a "subsection" of a list, and lists that are accidentally or purposefully broken up, but basically look like the first one (like in [produce](https://en.wiktionary.org/wiki/produce]).

Finally, we now look for lines in the list that look like it starts a POS-section. When a POS-section start is detected, active_pos is assigned that pos and each value fed into "sounds" gets a "pos"-field with that pos-value.

That pos-value is then used to filter away incompatible entries when they are merged into POS-section main data in merge_base() in parse_languages(). `If this entry has a "pos"-value that is not the "pos"-value of the current receiving data, then don't merge it`. The "pos"-field is also removed at the same time, so they don't hang around unnecessarily.
